[Wayland] Versioned interfaces ignored when compiling

### Tested versions

4.3 stable

### System information

Linux - Godot 4.3

### Issue description

Wayland interface version values should be used to control when extra fields are available, or not.

The below build errors can be avoided by checking interface version vs available fields.  Control code inclusion/exclusion based on interface version.
```
| platform/linuxbsd/wayland/wayland_thread.h:667:4: error: field designator 'preferred_buffer_scale' does not refer to any field in type 'const struct wl_surface_listener'
|   667 |                 .preferred_buffer_scale = _wl_surface_on_preferred_buffer_scale,
|       |                 ~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
| platform/linuxbsd/wayland/wayland_thread.h:668:4: error: field designator 'preferred_buffer_transform' does not refer to any field in type 'const struct wl_surface_listener'
|   668 |                 .preferred_buffer_transform = _wl_surface_on_preferred_buffer_transform,
|       |                 ~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
| platform/linuxbsd/wayland/wayland_thread.h:704:4: error: field designator 'axis_relative_direction' does not refer to any field in type 'const struct wl_pointer_listener'
|   704 |                 .axis_relative_direction = _wl_pointer_on_axis_relative_direction,
|       |                 ~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
| 3 errors generated.
```

### Steps to reproduce

Build 4.3 stable on Yocto scarthgap.

### Minimal reproduction project (MRP)

build issue

@Riteo With this change it produces the same error as reported.

```
$ git diff
diff --git a/recipes-graphics/godot/files/0001-Enable-build-config-wayland-yes-use_sowrap-no.patch b/recipes-graphics/godot/files/0001-Enable-build-config-wayland-yes-use_sowrap-no.patch
index d963a74..44ab580 100644
--- a/recipes-graphics/godot/files/0001-Enable-build-config-wayland-yes-use_sowrap-no.patch
+++ b/recipes-graphics/godot/files/0001-Enable-build-config-wayland-yes-use_sowrap-no.patch
@@ -19,7 +19,7 @@ index ce8a53a856..2638bec065 100644
  #include "wayland/egl_manager_wayland.h"
  #include "wayland/egl_manager_wayland_gles.h"
 +#ifndef SOWRAP_ENABLED
-+#include <wayland-egl.h>
++#include <wayland-egl-core.h>
 +#endif
  #endif
  
diff --git a/recipes-graphics/godot/godot_4.3.bb b/recipes-graphics/godot/godot_4.3.bb
index 8570f38..7c0b996 100644
--- a/recipes-graphics/godot/godot_4.3.bb
+++ b/recipes-graphics/godot/godot_4.3.bb
@@ -29,7 +29,6 @@ SRC_URI = " \
     git://github.com/godotengine/godot.git;protocol=https;lfs=0;branch=master \
     file://0001-Enable-build-config-wayland-yes-use_sowrap-no.patch \
     file://0001-enable-clang.patch \
-    file://0001-wayland-remove-fields.patch \
 "
 
 S = "${WORKDIR}/git"
```

```
| In file included from platform/linuxbsd/wayland/display_server_wayland.h:36:
| platform/linuxbsd/wayland/wayland_thread.h:667:4: error: field designator 'preferred_buffer_scale' does not refer to any field in type 'const struct wl_surface_listener'
|   667 |                 .preferred_buffer_scale = _wl_surface_on_preferred_buffer_scale,
|       |                 ~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
| platform/linuxbsd/wayland/wayland_thread.h:668:4: error: field designator 'preferred_buffer_transform' does not refer to any field in type 'const struct wl_surface_listener'
|   668 |                 .preferred_buffer_transform = _wl_surface_on_preferred_buffer_transform,
|       |                 ~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
| platform/linuxbsd/wayland/wayland_thread.h:704:4: error: field designator 'axis_relative_direction' does not refer to any field in type 'const struct wl_pointer_listener'
|   704 |                 .axis_relative_direction = _wl_pointer_on_axis_relative_direction,
|       |                 ~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

--- a/platform/linuxbsd/wayland/wayland_thread.h
+++ b/platform/linuxbsd/wayland/wayland_thread.h
@@ -44,7 +44,7 @@
 #include <wayland-client-core.h>
 #include <wayland-cursor.h>
 #ifdef GLES3_ENABLED
-#include <wayland-egl.h>
+#include <wayland-egl-core.h>
 #endif
 #include <xkbcommon/xkbcommon.h>
 #endif // SOWRAP_ENABLED

